src/form.rs

```diff
@@ -77,7 +77,7 @@ impl Form for MetaForm {
 pub enum CompactForm {}
 
 impl Form for CompactForm {
-	type String = UntrackedSymbol<&'static str>;
-	type TypeId = UntrackedSymbol<AnyTypeId>;
```

Comment:
Can you elaborate why `AnyTypeId` is not suitable here? We introduced it instead of our own `crate::type_id::TypeId` in order to counteract some unlimited recursion for some special types that recurse in their signature. E.g.
```rust
enum Tree<T> {
    Node{ lhs: Box<Self>, rhs: Box<Self> },
    Leaf(T),
}
```
Where we could in theory use `Box<Tree>` as `T` which screwed up some of our algorithms back then. Maybe we could also fix it by better designing those.
The nice property behind `AnyTypeId` is that it is a simple number basically given by the compiler.

Reply:
Actually you are right. It is not necessary to remove the AnyTypeId.
I'll (force) push a new version which does not changes this.